src/registries/auto_completes/auto_complete_registry.ts

```diff
@@ -23,6 +23,7 @@ export interface AutoCompleteProvider {
   proposals: AutoCompleteProposal[];
   selectProposal(text: string): void;
   autoSelectFirstProposal: boolean;
+  hideAutoCompleteBtn: boolean;
```

Comment:
I don't think it's a good name to mention the autocomplete button as it's 100% UI related while this object is merely a data structure. Also, I find the conditions it appears in are not clear. Maybe something like `canBeToggled`/`isHid(e)able`/`isFoldable`  ?